Make the platform name translatable

OmarIthawi opened the pull request:
### Description

TASK: [Edraak platform name in Arabic](https://app.asana.com/0/256944342521670/304473022051932)

This is to make the edX (and in my case Edraak) name and it's derivatives (like edX Studio) translatable. The current alternative is to hard code it everywhere in Arabic using the configuration repo.

After [digging in every edX repo](https://gist.github.com/OmarIthawi/c0d9045e4c0cab58b331c6bb8585b661), I haven't found the platform name in any meaningful place except for the platorm. At least not much to worry about!

### Testing
- [x] [Initial discussion on Slack](https://openedx.slack.com/archives/C4BTM66AW/p1498021293802098) with the Open edX team
- [x] Make da tests pass (except two unrelated failures, no?)!
- [x] Ensure backward compatibility when the `PLATFORM_NAME` is present in `TOKEN_ENVS`, the translation is effectively disabled!
- [x] Get engineering/product feedback from @edX
- [ ] Review all (the important) lms and cms pages
- [ ] Add few test cases (let's get some feedback first!)
- [ ] Add support for the feature in the [configuration repo](https://github.com/edx/configuration). Especially for `ANALYTICS_DASHBOARD_NAME` to address @UmanShahzad comment. [After Review and Approval]

### Out of Scope
 - This PR doesn't try to make the microsites' `PLATFORM_NAME` configurable.

### Caveats
 - Fork maintainers interested in this feature still have to change the platform name between `_('edX')` in the Python files, but at least this PR layout the foundation for this.
 - Fork maintainers have to modify the configuration a bit by removing all the `PLATFORM_NAME` and likes from being inserted into the `*.json` file.

### Related PRs
- First attempt: https://github.com/Edraak/edx-platform/pull/63
- A second attempt: https://github.com/Edraak/edx-platform/pull/87/commits/0bd6bc59b398810643987bb79580f56ff11ab329
- Gave up beacause of errors and made it only Arabic: https://github.com/Edraak/edx-platform/pull/87/commits/ae3000398f3fbd732c9145bba04c30027c544747]
- Local PR (closed): https://github.com/Edraak/edraak-platform/pull/50

edx-webhook commented:
Thanks for the pull request, @OmarIthawi! I've created [OSPR-1806](https://openedx.atlassian.net/browse/OSPR-1806) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

OmarIthawi commented:
jenkins run a11y

OmarIthawi commented:
jenkins run python

UmanShahzad commented on lms/envs/common.py:
```diff
@@ -2803,7 +2803,7 @@
 ANALYTICS_DATA_TOKEN = ""
 ANALYTICS_DASHBOARD_URL = ""
-ANALYTICS_DASHBOARD_NAME = PLATFORM_NAME + " Insights"
+ANALYTICS_DASHBOARD_NAME = _('edX Insights')
```
I think this should be 
```python
ANALYTICS_DASHBOARD_NAME = _(PLATFORM_NAME + ' Insights')
```

I have an instance running under "Client Insights" which wouldn't be the case if we did this.

OmarIthawi replied:
Hey @UmanShahzad, this won't work because of the following:

 - `_(PLATFORM_NAME + ' Insights')` won't make it right into `gettext` catalogue ([ref](https://inventwithpython.com/blog/2014/12/20/translate-your-python-3-program-with-the-gettext-module/)). 
 - The `ANALYTICS_DASHBOARD_NAME` is still backward compatible thus your use case is still perfectly supported. If you pass it using the TOKEN_ENVS (which is the default behaviour in `configuration`) you'll get your value. I'll tag you in the `lms/envs/aws.py`.

OmarIthawi commented on lms/envs/aws.py:
```diff
@@ -587,7 +587,7 @@
 # Analytics Dashboard
 ANALYTICS_DASHBOARD_URL = ENV_TOKENS.get("ANALYTICS_DASHBOARD_URL", ANALYTICS_DASHBOARD_URL)
-ANALYTICS_DASHBOARD_NAME = ENV_TOKENS.get("ANALYTICS_DASHBOARD_NAME", PLATFORM_NAME + " Insights")
+ANALYTICS_DASHBOARD_NAME = ENV_TOKENS.get("ANALYTICS_DASHBOARD_NAME", ANALYTICS_DASHBOARD_NAME)
```
@UmanShahzad This is still *kina* backward compatible if you pass it through the configuration repo.

UmanShahzad replied:
@OmarIthawi Yeah basically one would have to override this variable if they weren't already.

OmarIthawi commented:
jenkins run python

OmarIthawi commented:
@haikuginger I believe the commit d940bbfd09db927f634c25ead226e965a5a8a910 (from #15400) have broken the tests in the `course_modes`. Which is affecting the tests on master.

I'm not completely sure about this, but `git bisect` suggested it.

OmarIthawi commented:
@felipemontoya I think you'll like this. Surprisingly, it passes all the tests (except for a minor seemingly unrelated failure)!

felipemontoya commented on lms/envs/common.py:
```diff
@@ -45,7 +45,7 @@
 ################################### FEATURES ###################################
 # The display name of the platform to be used in templates/emails/etc.
-PLATFORM_NAME = "Your Platform Name Here"
+PLATFORM_NAME = _('edX')
```
why not leave it as  _('Your Platform Name Here')

OmarIthawi replied:
So I should change every other occurrence of edX, right?

felipemontoya replied:
Every ocurrence of `edX` is a big thing, but settings.PLATFORM_NAME is used across the platform a lot. I would not like that this reverts back to 'edX' by default

OmarIthawi replied:
I mean within the context of my PR. So this includes:

 - `PLATFORM_NAME`
 - `ANALYTICS_DASHBOARD_NAME`
 - `STUDIO_NAME`
 - etc

OmarIthawi replied:
What do you think @felipemontoya

felipemontoya replied:
Now I understand. In that scope makes total sense.

OmarIthawi replied:
Done @felipemontoya

felipemontoya commented:
@OmarIthawi nice, it does not look traumatic at all

haikuginger commented:
@OmarIthawi, thanks for the heads-up about the test failures. 😊

OmarIthawi commented:
Just a gentle reminder @gsong :)

gsong commented:
@edx/educator-devs 🔔

iloveagent57 commented on common/djangoapps/third_party_auth/tests/test_views.py:
```diff
@@ -81,6 +81,7 @@ def test_signed_metadata(self, key_name, pub_key_starts_with):
         self._validate_signed_metadata(pub_key_starts_with=pub_key_starts_with)
 
     def test_secure_key_configuration(self):
+        pass
```
I don't think you need this here :)

OmarIthawi replied:
I don't think so either 😄  I wonder how did this slip throw!

OmarIthawi commented:
Hey @iloveagent57 thanks for passing by! I've addressed the comment. Please let me know if there's anything else that I should do. I haven't got an idea how to test this PR yet. 

If you think that the existing tests are enough I think it's ready to go!

iloveagent57 commented:
@OmarIthawi I'm going to setup a sandbox with your branch for testing.  Once it's up and everything looks good, I'll merge this PR.

iloveagent57 commented:
jenkins run lettuce

OmarIthawi commented:
Hey @feanil, I think you might want to take a look here since it might need some updates for the configuration repo.

OmarIthawi commented:
Thanks @iloveagent57

feanil commented:
Thanks for the heads up @OmarIthawi
